### PR TITLE
[Cache] Add CouchbaseCollectionAdapter compatibility with sdk 3.0.0

### LINKED ISSUE #39132: [Cache] Add compatibility with Couchbase 3.0
The version 3.0 of the couchbase extension breaks compatibility (as expected for a major bump) excluding on objects we are using in the Cache Component.

version 2.6 => https://docs.couchbase.com/php-sdk/2.6/managing-connections.html
version 3.0 => https://docs.couchbase.com/php-sdk/current/howtos/managing-connections.html

The symfony/cache's adapter is not compatible with version 3.0

/cc @ajcerezo

### PR DESCRIPTION
**Pull request by ajcerezo**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #39132
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/15764

Adding Couchbase SDK 3 support

### CODE REVIEW

**Comment by derrabus** [comment]
@ajcerezo Is this PR will work in progress?

**Comment by ajcerezo** [comment]
@derrabus Yes, I'm sorry about the delay, but I have had many problems, I am finishing the documentation and integration tests, I have it almost finished.

Thk!

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+    private static function isThreeVersion(): bool
+    {
+        return \extension_loaded('couchbase') && version_compare(phpversion('couchbase'), '3.0.0', '>=') && version_compare(phpversion('couchbase'), '4.0', '<');
+    }
+
+    private static function isTwoDotSixVersion(): bool
+    {
+        return \extension_loaded('couchbase') && version_compare(phpversion('couchbase'), '2.6.0', '>=') && version_compare(phpversion('couchbase'), '3.0', '<');
+    }
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]: Are those methods called anywhere?

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            try {
+                $result = $this->connection->remove($id);
+
+                if (null === $result->mutationToken()) {
+                    $idsErrors[] = $id;
+                }
+            } catch (DocumentNotFoundException $exception) {
+            }
+        }
+//        }
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]:
```suggestion
```

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    private static function isTwoDotSixVersion(): bool
+    {
+        return \extension_loaded('couchbase') && version_compare(phpversion('couchbase'), '2.6.0', '>=') && version_compare(phpversion('couchbase'), '3.0', '<');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]:
```suggestion
    protected function doFetch(array $ids): array
```

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php (hunk 1 above)
Comment (ajcerezo) [reply src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]: Thank you for your comment, it is already deleted

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php (hunk 2 above)
Comment (ajcerezo) [reply src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]: It is already deleted

File: src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php (hunk 3 above)
Comment (ajcerezo) [reply src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php]: Ok it is done

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ajcerezo.

**Pull request merged**